sdk/python/kfp/registry/registry_client.py

```diff
@@ -152,73 +157,139 @@ def _validate_tag(self, tag: str) -> None:
+            config = self._load_context(config_file, config)
+
+        matched = None
+        if self._is_known_host():
+            matched = re.match(_KNOWN_HOSTS_REGEX[self._known_host_key][0],
                                self._host)
-            if matched:
-                repo_resource_format = ('projects/'
-                                        '{project_id}/locations/{location}/'
-                                        'repositories/{repo_id}'.format_map(
-                                            _SafeDict(matched.groupdict())))
-            else:
-                raise ValueError(f'Invalid host URL: {self._host}.')
-            registry_endpoint = 'https://artifactregistry.googleapis.com/v1'
-            api_endpoint = f'{registry_endpoint}/{repo_resource_format}'
-            package_endpoint = f'{api_endpoint}/packages'
-            package_name_endpoint = f'{package_endpoint}/{{package_name}}'
-            tags_endpoint = f'{package_name_endpoint}/tags'
-            versions_endpoint = f'{package_name_endpoint}/versions'
-            config = {
-                'host':
-                    self._host,
-                'upload_url':
-                    self._host,
-                'download_version_url':
-                    f'{self._host}/{{package_name}}/{{version}}',
-                'download_tag_url':
-                    f'{self._host}/{{package_name}}/{{tag}}',
-                'get_package_url':
-                    f'{package_name_endpoint}',
-                'list_packages_url':
-                    package_endpoint,
-                'delete_package_url':
-                    f'{package_name_endpoint}',
-                'get_tag_url':
-                    f'{tags_endpoint}/{{tag}}',
-                'list_tags_url':
-                    f'{tags_endpoint}',
-                'delete_tag_url':
-                    f'{tags_endpoint}/{{tag}}',
-                'create_tag_url':
-                    f'{tags_endpoint}?tagId={{tag}}',
-                'update_tag_url':
-                    f'{tags_endpoint}/{{tag}}?updateMask=version',
-                'get_version_url':
-                    f'{versions_endpoint}/{{version}}',
-                'list_versions_url':
-                    f'{versions_endpoint}',
-                'delete_version_url':
-                    f'{versions_endpoint}/{{version}}',
-                'package_format':
-                    f'{repo_resource_format}/packages/{{package_name}}',
-                'tag_format':
-                    f'{repo_resource_format}/packages/{{package_name}}/tags/{{tag}}',
-                'version_format':
-                    f'{repo_resource_format}/packages/{{package_name}}/versions/{{version}}',
-            }
         else:
-            logging.info(f'load_config not implemented for host: {self._host}')
+            if 'regex' in config:
+                matched = re.match(config['regex'], self._host)
+
+        if not bool(matched):
```

Comment:
nit: instead of type-casting a possible `None` to `bool`, consider:
```suggestion
        if matched is None:
```

Reply:
Done.